diff --git a/CMakeLists.txt b/CMakeLists.txt
index a17573b..2ff8518 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,8 +39,6 @@ qt_feature_module_begin(
 include(configure.cmake)
 qt_feature_module_end(NO_MODULE)
 
-set(WITH_LTM OFF CACHE BOOL "Build with support for libtommath" FORCE)
-add_subdirectory(3rdparty/libtomcrypt)
 add_subdirectory(qsqlcipher)
 
 if(NOT CMAKE_PROJECT_NAME STREQUAL "QtBase" AND NOT CMAKE_PROJECT_NAME STREQUAL "Qt")
diff --git a/qsqlcipher/CMakeLists.txt b/qsqlcipher/CMakeLists.txt
index 4af778e..c037978 100644
--- a/qsqlcipher/CMakeLists.txt
+++ b/qsqlcipher/CMakeLists.txt
@@ -16,6 +16,7 @@ qt_internal_add_plugin(QSQLCipherPlugin
         Qt::Core
         Qt::CorePrivate
         Qt::SqlPrivate
+    STATIC
 )
 
 ## Scopes:
@@ -35,6 +36,10 @@ if(QT_FEATURE_system_sqlite)
     qt_internal_force_macos_intel_arch(QSQLCipherPlugin)
 endif()
 
+if (NOT LIBTOMCRYPT_INCLUDE_DIR)
+    set(LIBTOMCRYPT_INCLUDE_DIR ../3rdparty/libtomcrypt/src/headers)
+endif()
+
 qt_internal_extend_target(QSQLCipherPlugin CONDITION NOT QT_FEATURE_system_sqlite
     SOURCES
         ../3rdparty/sqlcipher/sqlite3.c
@@ -53,9 +58,6 @@ qt_internal_extend_target(QSQLCipherPlugin CONDITION NOT QT_FEATURE_system_sqlit
         SQLITE_OMIT_COMPLETE
     INCLUDE_DIRECTORIES
         ../3rdparty/sqlcipher
-        ../3rdparty/libtomcrypt/src/headers
-    LIBRARIES
-        libtomcrypt
 )
 
 qt_internal_extend_target(QSQLCipherPlugin CONDITION CMAKE_BUILD_TYPE STREQUAL Release AND NOT QT_FEATURE_system_sqlite
@@ -101,3 +103,10 @@ qt_internal_extend_target(QSQLCipherPlugin CONDITION NOT QT_FEATURE_thread AND N
     DEFINES
         SQLITE_THREADSAFE=0
 )
+
+qt_internal_extend_target(QSQLCipherPlugin CONDITION NOT QT_FEATURE_system_sqlite AND LIBTOMCRYPT_INCLUDE_DIR
+    INCLUDE_DIRECTORIES
+        ${LIBTOMCRYPT_INCLUDE_DIR}
+    LIBRARIES
+        libtomcrypt
+)
